### PR TITLE
SDI-438: 🔒️ Upgrade spring security for CVE

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManagerTest.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ class DependencyManagementPluginManagerTest : GradleBuildTest() {
           id("uk.gov.justice.hmpps.gradle-spring-boot") version "0.1.0"
         }
         dependencies {
-            implementation("org.springframework.boot:spring-boot-starter-web:2.7.2");
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManagerTest.kt:18]: this didn't need to be pinned to a version anyway

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt [hunk 2]
```diff
@@ -12,8 +12,10 @@ class DependencyManagementPluginManager(override val project: Project) : PluginM
     applyDependencyManagementBom(project)
     // overriding snakeyaml for https://nvd.nist.gov/vuln/detail/CVE-2022-25857 - brought in by org.springframework.boot:spring-boot-dependencies:2.7.2
     // overriding snakeyaml for https://nvd.nist.gov/vuln/detail/CVE-2022-38751 - brought in by org.springframework.boot:spring-boot-dependencies:2.7.2
-    project.extensions.extraProperties["snakeyaml.version"] = "1.32"
+    project.extensions.extraProperties["snakeyaml.version"] = "1.33"
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt:15]: latest spring boot still pulls in 1.30

**Review by andrewrlee — APPROVED** [review]
LGTM

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**